app/models/audit/AuditTaskTable.scala

```diff
@@ -321,183 +339,172 @@ object AuditTaskTable {
   }
 
   /**
-   * get a new task for the user
+   * Get a new task for the user. This is called if the user is not already assigned a region.
    *
    * Reference for creating java.sql.timestamp
    * http://stackoverflow.com/questions/308683/how-can-i-get-the-current-date-and-time-in-utc-or-gmt-in-java
    * http://alvinalexander.com/java/java-timestamp-example-current-time-now
    *
-   * Subqueries in Slick
-   * http://stackoverflow.com/questions/14425844/why-does-slick-generate-a-subquery-when-take-method-is-called
-   * http://stackoverflow.com/questions/14920153/how-to-write-nested-queries-in-select-clause
-   *
    * @param user User ID.
    * @return
    */
   def selectANewTask(user: UUID): NewTask = db.withSession { implicit session =>
     val timestamp: Timestamp = new Timestamp(Calendar.getInstance(TimeZone.getTimeZone("UTC")).getTime.getTime)
 
     val tasksCompletedByUser = for {
-      u <- users.filter(_.userId === user.toString)
-      at <- completedTasks if at.userId === u.userId
-    } yield (u.username.?, at.streetEdgeId.?)
+      _user <- users.filter(_.userId === user.toString)
+      _task <- completedTasks if _task.userId === _user.userId
+    } yield (_user.username.?, _task.streetEdgeId.?)
 
-    // gets list of streets that user has not audited, takes 100, then picks one of those at random to assign
-    val edges = for {
-      (e, c) <- streetEdges.leftJoin(tasksCompletedByUser).on(_.streetEdgeId === _._2)
-      if c._1.isEmpty && !e.deleted
-    } yield e
+    println("Called selectANewTask:userid - no region assigned")
+    // Gets list of streets that user has not audited, takes 100, then picks the one with highest priority to assign
+    // TODO: Remove randomness once real-time updates of priority have been implemented.
+    val edges = (for {
+      (_edge, _task) <- streetEdgesWithoutDeleted.leftJoin(tasksCompletedByUser).on(_.streetEdgeId === _._2)
+      if _task._1.isEmpty
+    } yield _edge).take(100)
 
-    val edgesWithCompCount = (for {
-      (se, ac) <- edges.innerJoin(streetCompletionCounts).on(_.streetEdgeId === _._1)
-    } yield (se.streetEdgeId, se.geom, se.x1, se.y1, se.x2, se.y2, timestamp, ac._2)).take(100).list
+    val edgesWithCompletionCountAndPriority = for {
+      se <- edges
+      scc <- streetCompletionCounts if se.streetEdgeId === scc._1
+      sep <- StreetEdgePriorityTable.streetEdgePriorities if scc._1 === sep.streetEdgeId
+    } yield (se.streetEdgeId, se.geom, se.x1, se.y1, se.x2, se.y2, timestamp, scc._2, sep.priority)
+    assert(edgesWithCompletionCountAndPriority.list.nonEmpty)
```

Comment:
Not sure if we want the assert there, but if the list is empty, then the following code isn't going to work anyway... The code should always work as long as a user has not audited all of DC itself. I really am not sure what to do here, I put in asserts because that is what was done in one of these functions (written by someone else).

I had meant to ask about this: In the case where there are asserts right now, what should we do if these things fail? They really should never fail, but what protocol should we use in these situations? The clearest case is in the selectANewTask function. We should _always_be able to find a task. But at some point we need to call `.head` or `.get`, and if one of those fails, our code crashes. So I think it is really bad practice to leave a `.head` or `.get` without any sort of error checking/handling/acknowledgement even. What do you think we should do in those cases?